engine/src/main/java/com/redhat/cloud/notifications/transformers/BaseTransformer.java

```diff
@@ -24,33 +26,37 @@ public class BaseTransformer {
     public static final String TIMESTAMP = "timestamp";
 
     /**
-     * Transforms the given action into a {@link JsonObject}.
-     * @param action the {@link Action} to transform.
-     * @return a {@link JsonObject} containing the given action.
+     * Transforms the given event into a {@link JsonObject}.
+     * @param event the {@link Event} to transform.
+     * @return a {@link JsonObject} containing the given event.
      */
-    public JsonObject toJsonObject(final Action action) {
+    public JsonObject toJsonObject(final Event event) {
         JsonObject message = new JsonObject();
 
-        message.put(ACCOUNT_ID, action.getAccountId());
-        message.put(APPLICATION, action.getApplication());
-        message.put(BUNDLE, action.getBundle());
-        message.put(CONTEXT, JsonObject.mapFrom(action.getContext()));
-        message.put(EVENT_TYPE, action.getEventType());
-        message.put(
-            EVENTS,
-            new JsonArray(
-                action.getEvents().stream().map(
-                    event -> Map.of(
-                        METADATA, JsonObject.mapFrom(event.getMetadata()),
-                        PAYLOAD, JsonObject.mapFrom(event.getPayload())
-                    )
-                ).collect(Collectors.toList())
-            )
-        );
-        message.put(ORG_ID, action.getOrgId());
-        message.put(TIMESTAMP, action.getTimestamp().toString());
+        message.put(BUNDLE_DISPLAY_NAME, event.getBundleDisplayName());
```

Comment:
From an RHCLOUD-22667 perspective, the app display name is missing here. But as discussed in other comments, we'll probably add more fields as well.